packages/compiler-sfc/src/script/resolveType.ts

```diff
@@ -1688,7 +1688,7 @@ export function inferRuntimeType(
       case 'TSUnionType':
         return flattenTypes(ctx, node.types, scope)
```

Comment:
Should this also apply to UnionTypes?

Reply:
Yes, I've also encountered an issue with union types during testing:
- When `type A = 'a' | 'b'`, `resolveType` correctly returns `['String']`.
- When `type A = number | string`, `resolveType` returns `['Number', 'String']`, but in `TypeScript`, the type of `A` should be `"toString" | "valueOf"`.

Given this, I believe this might be another issue, and simply adding a parameter might not fix it. Do you think fixing this issue is necessary? If necessary, I'll try to address it.